core/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -341,12 +340,13 @@ public void onTimeout(TimeValue timeout) {
                     }
                     // use null timeout as we use timeout on the latchwait
                 }, MasterNodeChangePredicate.INSTANCE, null);
-            }
-
-            try {
-                latch.await(DiscoverySettings.INITIAL_STATE_TIMEOUT_SETTING.get(settings).millis(), TimeUnit.MILLISECONDS);
-            } catch (InterruptedException e) {
-                throw new ElasticsearchTimeoutException("Interrupted while waiting for initial discovery state");
+                try {
+                    if (latch.await(DiscoverySettings.INITIAL_STATE_TIMEOUT_SETTING.get(settings).millis(), TimeUnit.MILLISECONDS) == false) {
+                        logger.warn("timed out while waiting for initial timeout: {}", DiscoverySettings.INITIAL_STATE_TIMEOUT_SETTING.get(settings));
```

Comment:
The message doesn't really make sense because it's missing the "discovery state" part, right now it says it timed out waiting for initial timeout :)